### PR TITLE
feat: add shipped-in release lookup script and canonical tracking hook

### PR DESCRIPTION
**Pull request by panghy**
## Summary

AGENTS.md § Release Process requires every intentd/cloudlands-fe workspace to monitor until the change ships in a cloudlands-fe alpha and report the carrying version, but offered no supported *how* — each agent hand-wrote its own polling hook (which repo to list, how to decide "tag contains my commit", where the manifest lives, the intentd pin-bump case). This adds one tested script plus the canonical hook that calls it, and shrinks the prose to "run this".

Retrospective rung 3 (make the right path discoverable via a script). Rung 1/2 are infeasible: shipping is an external asynchronous event that nothing in-repo can type-check or CI-gate.

## Changes

- `scripts/shipped-in.sh <intentd|cloudlands-fe> <sha> [--limit N]`
  - Lists the newest N (default 10) `vX.Y.Z` releases on `intent-hq/cloudlands-releases`.
  - cloudlands-fe: `gh api repos/intent-hq/cloudlands-fe/compare/<sha>...<tag>`; a hit is `ahead|identical` only (`behind`/`diverged` never count).
  - intentd: reads `intentdVersion` from each tag's `release-manifest.json` and compares `<sha>...v<version>` on `intent-hq/intentd` (deduplicated per pinned version).
  - Prints the oldest carrying tag as `<tag> intentdVersion=<v>` and exits 0; exits 3 (empty stdout) while nothing carries the commit yet; 2 on usage errors; 1 on `gh`/API failure.
- `scripts/shipped-in.test.sh` — hermetic, stubbed `gh` on PATH (same convention as `dev-status.test.sh`); wired into the `shell-tests` CI job.
- `make shipped-in COMPONENT=... SHA=... [LIMIT=N]` (listed in `make help`).
- AGENTS.md "Track shipped work" bullet rewritten around the script with a canonical `ws.hook.schedule` snippet (dispatch on exit 0, keep waiting on exit 3, throw otherwise), mirroring the sandbox health-wait snippet.
- One-line pointer in `docs/RELEASING.md` § Gotchas.

## Verification

| Claim | Command | Result |
|---|---|---|
| Unit tests | `bash scripts/shipped-in.test.sh` | pass (hit-first, hit-later/oldest wins, no hit → 3, behind/diverged miss, compare 404 → 1, gh failure → 1 not 3, intentd manifest path + dedup, missing manifest → 1, usage → 2, `--limit` forwarding) |
| Doc gates | `make docs-check`, `node scripts/check-doc-links.mjs`, `node scripts/check-root-hygiene.mjs`, `bash scripts/docs-check.test.sh` | all pass |
| Not yet shipped | `scripts/shipped-in.sh cloudlands-fe 66e5cb92c4858dd3119b1d12d369493bec69dfb2` (cloudlands-fe#2260) | exit 3, empty stdout |
| fe hit | `scripts/shipped-in.sh cloudlands-fe 5b466957241fe6e6e3533294e6a86bf61083557c` (v2.144.0 `feSha`) | `v2.144.0 intentdVersion=0.9.38`, exit 0 |
| intentd hit | `scripts/shipped-in.sh intentd <v0.9.37 commit> --limit 6` | `v2.143.0 intentdVersion=0.9.38`, exit 0, ~7 s |
| Hook path | `ws.host.exec({ command: "scripts/shipped-in.sh", ... })` | same exit codes as above |

Note: `make shipped-in` surfaces the script's exit 3 as make `Error 3`, so hooks invoke the script directly (documented in the Makefile comment and AGENTS.md snippet).

### CODE REVIEW

**Review by panghy — COMMENTED** [review]
Comment-only review of f538eeb. Requested unit/doc checks pass; live read-only FE hit/no-hit and intentd pin lookup pass. GitHub compare base...head direction is correct (ahead/identical mean inclusion); gh 2.45.0 accepts --exclude-drafts and release download --output -. The restricted test PATH resolves gh only to the stub. The AGENTS.md hook and nested code parse and correctly dispatch/continue/throw for exits 0/3/1. Two error-propagation gaps below violate the promised exit-1 failure contract. Note: the second reproduction injects a gh command that emits a tag before failing; I did not observe that failure mode from the live GitHub CLI. The documented first release is only the oldest hit within the configured scan window, as explained in the script and Makefile. No approval or merge action taken.

File: scripts/shipped-in.sh [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+  fi
+  carries "$status" && first_hit=$tag
+done
+
+if [[ -z "$first_hit" ]]; then
+  echo "shipped-in: $sha is not carried by the newest ${#tags[@]} release(s) on $releases_repo (newest ${tags[0]})" >&2
+  exit 3
+fi
+
+printf '%s intentdVersion=%s\n' "$first_hit" "$(manifest_version "$first_hit")"
```
Comment (panghy) [inline scripts/shipped-in.sh]: [P2] Propagate the winning manifest failure before printing success. Repro: release list returns v2.3.0, compare returns ahead, and release download exits 1 (the same happens for malformed/missing intentdVersion JSON). Observed exit 0, stdout "v2.3.0 intentdVersion=", and an error on stderr. The command substitution exits 1, but printf succeeds, so set -e does not propagate the failure and the canonical hook dispatches success despite incomplete output. Read the winning version into a checked standalone assignment before printf. Add FE hit + missing manifest and malformed manifest tests asserting exit 1 and empty stdout; the existing missing-manifest test only exercises the earlier intentd assignment path. This is the original reviewed commit; the coordinator has been notified and is preparing the fix.

File: scripts/shipped-in.sh [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+    ahead | identical) return 0 ;;
+    *) return 1 ;;
+  esac
+}
+
+mapfile -t tags < <(
+  gh release list --repo "$releases_repo" --limit "$limit" --exclude-drafts \
+    --json tagName --jq '.[].tagName' 2>/dev/null |
+    grep -E '^v[0-9]+\.[0-9]+\.[0-9]+' || true
+)
```
Comment (panghy) [inline scripts/shipped-in.sh]: [P2] Preserve gh release-list failure independently of collected tags. An injected gh stub that prints v2.3.0 then exits 1 is accepted here: with ahead and a valid manifest for that tag, the script exits 0 and prints a carrying release instead of returning the promised exit 1. Both the process substitution and the blanket || true discard the pipeline status; checking only the tag count catches failures only when stdout is empty. Capture gh release-list output in a checked assignment first, then filter/mapfile the successful result. Add a regression where release list emits one valid tag and fails, asserting exit 1, empty stdout, and no compare/download calls. This is defensive error-contract coverage; live gh JSON may buffer output, so I am not claiming an observed GitHub pagination failure.

**Review by panghy — COMMENTED** [review]
Re-verified current head c4424c010ff81ea8c4aa2d65dc50477f65da3c07. Both P2 findings are fixed: 98ff1fa propagates winning manifest failures before printing; c4424c0 checks release-list exit status before filtering. Requested shell tests, make docs-check, documentation links, bash syntax, and diff whitespace checks all pass. Independent fault injection for both components confirms missing manifests, invalid JSON, release-list output followed by failure, and manifest output followed by failure all return exit 1 with empty stdout. A failed release list makes no compare/download calls. Live FE and intentd success lookups still print the expected carrying tags. No remaining blocking findings in the scoped code review. This is a comment-only verification follow-up, not a GitHub approval or merge authorization.

File: scripts/shipped-in.sh [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+
+carries() {
+  case "$1" in
+    ahead | identical) return 0 ;;
+    *) return 1 ;;
+  esac
+}
+
+mapfile -t tags < <(
+  gh release list --repo "$releases_repo" --limit "$limit" --exclude-drafts \
```
[the remark this answers is not in the document]
Comment (panghy) [reply scripts/shipped-in.sh]: Fixed in 63e346c: confirmed `alpha`, `beta`, and `stable` rolling releases exist on the distribution repo. The script now over-fetches (`--limit N+10`), filters to `vX.Y.Z` tags, and takes the newest N of those. Regression test interleaves channel releases and asserts `--limit 2` scans exactly two versioned tags and never compares channel tags.

File: scripts/shipped-in.sh (hunk 1 above)
Comment (panghy) [reply scripts/shipped-in.sh]: Fixed in 98ff1fa: the winning tag manifest is resolved in a checked standalone assignment before printing. Regression tests cover an FE hit with a missing and with a malformed manifest (exit 1, empty stdout).

File: scripts/shipped-in.sh (hunk 2 above)
Comment (panghy) [reply scripts/shipped-in.sh]: Fixed in c4424c0: the release listing is captured in a checked assignment (`|| fail` on nonzero gh exit) and filtered separately. Regression test has the stub emit a valid tag and then exit 1 (exit 1, empty stdout, no compare or download performed).
